source/common/io/io_uring_worker_impl.cc

```diff
@@ -611,5 +638,41 @@ void IoUringServerSocket::submitWriteOrShutdownRequest() {
   }
 }
 
+IoUringClientSocket::IoUringClientSocket(os_fd_t fd, IoUringWorkerImpl& parent,
+                                         Event::FileReadyCb cb, uint32_t write_timeout_ms,
+                                         bool enable_close_event)
+    : IoUringServerSocket(fd, parent, cb, write_timeout_ms, enable_close_event) {}
+
+void IoUringClientSocket::connect(const Network::Address::InstanceConstSharedPtr& address) {
+  // Reuse read request since there is no read on connecting and connect is cancellable.
+  ASSERT(read_req_ == nullptr);
+  read_req_ = parent_.submitConnectRequest(*this, address);
+}
+
+void IoUringClientSocket::onConnect(Request* req, int32_t result, bool injected) {
+  IoUringSocketEntry::onConnect(req, result, injected);
+  ASSERT(!injected);
+  ENVOY_LOG(trace, "onConnect with result {}, fd = {}, injected = {}, status_ = {}", result, fd_,
+            injected, status_);
+
+  read_req_ = nullptr;
+  // Socket may be closed on connecting like binding error. In this situation we may not callback
+  // on connecting completion.
+  if (status_ == Closed) {
+    if (write_or_shutdown_req_ == nullptr && read_cancel_req_ == nullptr &&
+        write_or_shutdown_cancel_req_ == nullptr) {
+      closeInternal();
+    }
+    return;
+  }
+
+  if (result == 0) {
+    enableRead();
```

Comment:
maybe we can add unittest for the case the connection failed and the read won't be enabled.

Reply:
Added, thanks.